client/src/App.tsx

```diff
@@ -0,0 +1,46 @@
+import './App.css';
+import React, {useEffect, useState} from 'react';
+import RenderReviewList from './components/RenderReviewList';
+import SignInWithSlack from './components/SignInWithSlack';
+import Navbar from './components/Navbar';
+import ChannelFilter from './components/ChannelFilter';
+import {logError} from './services/log';
+import {fetchData} from './services/fetch';
+
+function App() {
+    const queryString = new URLSearchParams(window.location.search);
+    const [status, setStatus] = useState(queryString.get('status') || 'pending');
+    const [user, setUser] = useState(null);
+    const [selectedChannel, setSelectedChannel] = useState(queryString.get('channel') || 'all');
+
+    const handleNavBarClick = (value: string) => {
+        setStatus(value);
+    };
+
+    useEffect(() => {
+        fetchData('/api/user')
+            .then((data) => setUser(data))
```

Comment:
Since `fetchData` is async, there is a possibility that the component can unmount before the API call resolves. However, `setUser(data)` would still be called and throw an error. To avoid this, we need to use the `useEffect` cleanup function to ensure that we only perform state modifying actions if this hook has not been cleaned up - https://react.dev/reference/react/useEffect#fetching-data-with-effects